api/v1alpha1/linodeobjectstoragebucket_types.go

```diff
@@ -43,6 +43,11 @@ type LinodeObjectStorageBucketSpec struct {
 	// +optional
 	// +kubebuilder:default=0
 	KeyGeneration *int `json:"keyGeneration,omitempty"`
+
+	// secretType sets the type for the access-keys secret that will be generated by the controller.
+	// +optional
+	// +kubebuilder:default=addons.cluster.x-k8s.io/resource-set
+	SecretType string `json:"secretType,omitempty"`
```

Comment:
What do you think about defaulting `SecretType` to `Opaque` to support the general use case of provisioning buckets? Otherwise, the formatting of the secret data could throw users off.

Reply:
So Evan and I had a chat about this and the OBJ use case is going to be for the etcd backup almost 90% of the times. @eljohnson92 thoughts? Should we keep this as is or swap the defaults?